### DISCUSSION TITLE
How can I get Vitest to work with TypeORM?

### DISCUSSION THREAD

**Discussion by ChristianPavilonis** [vitest-dev/vitest#2751] [Q&A] [answered]
I had it working a while ago but I came back today now I'm getting all sorts of issues.

Now when run vitest I get this error:
```
node_modules/typeorm/browser/platform/PlatformTools.js:1 seems to be an ES Module but shipped in a CommonJS package. You might want to create an issue to the package "typeorm" asking them to ship the file in .mjs extension or add "type": "module" in their package.json.
```

It asks me to inline the package like so:
```js
export default {
  test: {
    deps: {
      inline: [
        "typeorm"
      ]
    }
  }
}
```

If I do I end up with this error:
```
TypeError: Class extends value undefined is not a constructor or null
 ❯ Object.<anonymous> src/connection/Connection.ts:10:33
```

I tried to use the swc plugin as recommended in this issue: https://github.com/vitest-dev/vitest/issues/708#issuecomment-1118628479 , but I get the same error.

Are there any configurations out there that work? or do I need to switch to Jest?

**Comment by ChristianPavilonis** [comment, marked as answer]
Never mind... I realize I was imported a decorator from 'typeorm/browser'

**Comment by carlosmoliv** [comment]
I'm having the same issue, could you take a look please?

Here is my question on stack overflow: https://stackoverflow.com/questions/76157680/how-to-resolve-syntaxerror-cannot-use-import-statement-outside-a-module-when-ru

**Reply by f-llewellyn** [reply]
Hi there. Did you ever find a resolution to this?
